### PR TITLE
feat: added default values & missing fields to Upstream

### LINKED ISSUE #1770: there is no default status code when configuring the health check through the Dashboard
# Feature request

## Please describe your feature

When configuring the health check in the Dashboard, if the status code information is not set, there is no default status code. When configuring the health check through the `Admin API`, there will be a status code by default.

1. Configure the health check through the Dashboard
Create a route and do not set the `HTTP Status` in the passive health check:

截屏2021-04-14 下午6 47 45

The etcd data is as follows:

```yaml
{
    "id":"350421980362573985",
    "create_time":1618397376,
    "update_time":1618397376,
    "uris":[
        "/*"
    ],
    "name":"dashboard-healthcheck",
    "methods":[
        "GET",
        "HEAD",
        "POST"
    ],
    "upstream":{
        "nodes":[
            {
                "host":"127.0.0.1",
                "port":80,
                "weight":1
            }
        ],
        "timeout":{
            "connect":6,
            "read":6,
            "send":6
        },
        "type":"roundrobin",
        "checks":{
            "active":{
                "healthy":{
                    "interval":10,
                    "successes":5
                },
                "host":"localhost",
                "http_path":"/hello",
                "port":1980,
                "timeout":2,
                "unhealthy":{
                    "http_failures":3,
                    "interval":10
                }
            },
            "passive":{
                "healthy":{
                    "successes":3
                },
                "unhealthy":{
                    "http_failures":3,
                    "tcp_failures":3
                }
            }
        },
        "pass_host":"pass"
    },
    "status":1
}
```

2. Configure the health check through the `Admin API`, and do not set the `HTTP Status` in the passive health check.

Create a route:

```shell
curl http://127.0.0.1:9080/apisix/admin/routes/1 -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X PUT -d '
{
    "uri": "/admin-api",
    "upstream": {
         "nodes": {
            "127.0.0.1:1980": 1,
            "127.0.0.1:1970": 1
        },
        "type": "roundrobin",
        "checks": {
            "active": {
                "timeout": 5,
                "http_path": "/status",
                "host": "foo.com",
                "healthy": {
                    "interval": 2,
                    "successes": 1
                },
                "unhealthy": {
                    "interval": 1,
                    "http_failures": 2
                }
            },
            "passive": {
                "healthy": {
                    "successes": 3
                },
                "unhealthy": {
                    "http_failures": 3,
                    "tcp_failures": 3
                }
            }
        }
    }
}'
```

The etcd data is as follows:

```json
{
    "update_time":1618397719,
    "id":"1",
    "create_time":1618394497,
    "uri":"/admin-api",
    "status":1,
    "priority":0,
    "upstream":{
        "type":"roundrobin",
        "scheme":"http",
        "checks":{
            "active":{
                "healthy":{
                    "interval":2,
                    "successes":1,
                    "http_statuses":[
                        200,
                        302
                    ]
                },
                "concurrency":10,
                "http_path":"/status",
                "https_verify_certificate":true,
                "timeout":5,
                "unhealthy":{
                    "http_failures":2,
                    "tcp_failures":2,
                    "interval":1,
                    "timeouts":3,
                    "http_statuses":[
                        429,
                        404,
                        500,
                        501,
                        502,
                        503,
                        504,
                        505
                    ]
                },
                "type":"http",
                "host":"foo.com"
            },
            "passive":{
                "healthy":{
                    "http_statuses":[
                        200,
                        201,
                        202,
                        203,
                        204,
                        205,
                        206,
                        207,
                        208,
                        226,
                        300,
                        301,
                        302,
                        303,
                        304,
                        305,
                        306,
                        307,
                        308
                    ],
                    "successes":3
                },
                "unhealthy":{
                    "timeouts":7,
                    "http_statuses":[
                        429,
                        500,
                        503
                    ],
                    "http_failures":3,
                    "tcp_failures":3
                },
                "type":"http"
            }
        },
        "nodes":{
            "127.0.0.1:1980":1,
            "127.0.0.1:1970":1
        },
        "pass_host":"pass",
        "hash_on":"vars"
    }
}
```
## Describe the solution you'd like

I think that when configuring the health check through the Dashboard, there should be a default status code. Keep consistent with `Admin API`.

### PR DESCRIPTION
**Pull request by juzhiyuan**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [ ] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

Hi, this PR aims to add the **missing fields** to Upstream Form, according to schema_def. (https://github.com/apache/apisix/blob/master/apisix/schema_def.lua#L114) And all default value are added.

- [x] Concurrency
- [x] ...

fix #1770 

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [ ] Have you added corresponding test cases?
- [ ] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

File: web/src/components/Upstream/components/TLS.tsx [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+const CUSTOM_FIELD_KEY = "custom__enable_tls"
+
+const TLSComponent: React.FC<Props> = ({ form, readonly }) => {
+  const { formatMessage } = useIntl()
+
+  return (
+    <React.Fragment>
+      <Form.Item
+        label="启用 TLS"
```
Comment (LiteSun) [inline web/src/components/Upstream/components/TLS.tsx:34]: i18n.

File: web/src/components/Upstream/components/active-check/Healthy/HttpStatuses.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+const Component: React.FC<Props> = ({ readonly }) => {
+  const { formatMessage } = useIntl()
+
+  return (
+    <Form.List name={['checks', 'active', 'healthy', 'http_statuses']} initialValue={[200, 302]}>
+      {(fields, { add, remove }) => (
+        <>
+          <Form.Item
+            required
```
Comment (LiteSun) [inline web/src/components/Upstream/components/active-check/Healthy/HttpStatuses.tsx:36]: Does the form validation work when written this way?

File: web/src/components/Upstream/components/active-check/Host.tsx [hunk 3]
```diff
@@ -26,30 +26,30 @@ const Component: React.FC<Props> = ({ readonly }) => {
...
         name={['checks', 'active', 'host']}
         rules={[
           {
             required: true,
-            message: formatMessage({ id: 'page.upstream.step.input.healthyCheck.activeHost' }),
+            message: formatMessage({ id: 'component.upstream.fields.checks.active.host.required' }),
           },
           {
             pattern: new RegExp(
-              /(^([1-9]?\d|1\d{2}|2[0-4]\d|25[0-5])(\.(25[0-5]|1\d{2}|2[0-4]\d|[1-9]?\d)){3}$|^(?![0-9.]+$)([a-zA-Z0-9_-]+)(\.[a-zA-Z0-9_-]+){0,}$)/,
```
Comment (LiteSun) [inline web/src/components/Upstream/components/active-check/Host.tsx:43]: Any reference about this `RegExp ` change?

Comment (juzhiyuan) [reply web/src/components/Upstream/components/active-check/Host.tsx:43]: host_def_pat

Comment (juzhiyuan) [reply web/src/components/Upstream/components/active-check/Host.tsx:43]: https://github.com/apache/apisix/blob/master/apisix/schema_def.lua#L40

File: web/src/components/Upstream/components/active-check/Healthy/HttpStatuses.tsx (hunk 2 above)
Comment (juzhiyuan) [reply web/src/components/Upstream/components/active-check/Healthy/HttpStatuses.tsx:36]: all those options are optional, only for style effect.

image

**Comment by juzhiyuan** [comment]
CI passes on local

File: web/src/components/Upstream/UpstreamForm.tsx [hunk 4]
```diff
@@ -86,14 +87,18 @@ const UpstreamForm: React.FC<Props> = forwardRef(
...
         }
       } else {
         if (upstream_id) {
           requestAnimationFrame(() => {
-            form.setFieldsValue(list.find((item) => item.id === upstream_id));
+            const targetData = list.find((item) => item.id === upstream_id) as UpstreamComponent.ResponseData
+            if (targetData) {
+              form.setFieldsValue(transformUpstreamDataFromRequest(targetData));
+            } else {
+              // TODO: 提示 upstream_id 找不到想要的数据
```
Comment (imjoey) [inline web/src/components/Upstream/UpstreamForm.tsx:100]: @juzhiyuan should we use English here? Thanks.

File: web/src/components/Upstream/components/TLS.tsx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+            if (form.getFieldValue(["custom", "tls"]) === 'enable') {
+              return (
+                <React.Fragment>
+                  <Form.Item
+                    label={formatMessage({ id: 'component.upstream.fields.tls.client_cert' })}
+                    name={["tls", "client_cert"]}
+                    required
+                    rules={[{ required: true, message: "" }, { max: 64 * 1024 }, { min: 128 }]}
+                  >
+                    <Input.TextArea disabled={readonly} minLength={128} maxLength={64 * 1024} rows={5} placeholder="请输入客户端证书" />
```
Comment (LiteSun) [inline web/src/components/Upstream/components/TLS.tsx:72]: ditto

File: web/src/components/Upstream/components/TLS.tsx [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+                  </Form.Item>
+                  <Form.Item
+                    label={formatMessage({ id: 'component.upstream.fields.tls.client_key' })}
+                    name={["tls", "client_key"]}
+                    required
+                    rules={[{ required: true, message: "" }, { max: 64 * 1024 }, { min: 128 }]}
+                  >
+                    <Input.TextArea disabled={readonly} minLength={128} maxLength={64 * 1024} rows={5} placeholder="请输入客户端私钥" />
```
Comment (LiteSun) [inline web/src/components/Upstream/components/TLS.tsx:80]: ditto

File: web/src/components/Upstream/components/TLS.tsx (hunk 5 above)
Comment (imjoey) [inline web/src/components/Upstream/components/TLS.tsx:72]: @juzhiyuan The text of `placeholder` seems no i18n here. As expected?

File: web/src/components/Upstream/components/TLS.tsx (hunk 6 above)
Comment (imjoey) [inline web/src/components/Upstream/components/TLS.tsx:80]: ditto here.

File: web/src/components/Upstream/UpstreamForm.tsx (hunk 4 above)
Comment (juzhiyuan) [reply web/src/components/Upstream/UpstreamForm.tsx:100]: sure! I missed some notes, and will file a new PR to remove all Chinese.

Comment (moonming) [reply web/src/components/Upstream/UpstreamForm.tsx:100]: yes, English only

**Comment by moonming** [comment]
@juzhiyuan please use a better title of this PR

File: web/src/components/Upstream/components/passive-check/Type.tsx [hunk 7]
```diff
@@ -15,43 +15,55 @@
...
+            </Select>
+          </Form.Item>
+        </Col>
+      </Row>
     </Form.Item>
   )
 }
 
-export default PassiveCheckTypeComponent
+export default ActiveCheckTypeComponent
```
Comment (LiteSun) [inline web/src/components/Upstream/components/passive-check/Type.tsx:69]: wrong name.

**Review by imjoey — CHANGES_REQUESTED** [review]

File: web/src/components/Upstream/components/TLS.tsx (hunk 5 above)
Comment (juzhiyuan) [reply web/src/components/Upstream/components/TLS.tsx:72]: No, same as your last comment.

**Review by LiteSun — APPROVED** [review]

File: web/src/components/Upstream/components/TLS.tsx (hunk 5 above)
Comment (juzhiyuan) [reply web/src/components/Upstream/components/TLS.tsx:72]: Due to CI will take a long time to run, I would prefer to remove them in the new PR.

**Comment by juzhiyuan** [comment]
Hi @imjoey @LiteSun , thanks for your reviewing, just noticed there have some comments related to i18n (I missed them), I would prefer resolving them in the new PR, because GitHub Action will queue unexpected those days.

**Review by imjoey — APPROVED** [review]
/lgtm

Thanks @juzhiyuan .

**Review by membphis — APPROVED** [review]

**Review by moonming — APPROVED** [review]

File: web/src/components/Upstream/UpstreamForm.tsx (hunk 4 above)
Comment (juzhiyuan) [reply web/src/components/Upstream/UpstreamForm.tsx:100]: I just filed a PR to make the Upstream Logics more clear https://github.com/apache/apisix-dashboard/pull/1784, no need to check if target data lost any more.

**Pull request merged**